src/computability/halting.lean

```diff
@@ -161,7 +161,7 @@ open nat.partrec (code) nat.partrec.code computable
 theorem computable_iff {p : α → Prop} :
   computable_pred p ↔ ∃ f : α → bool, computable f ∧ p = λ a, f a :=
 ⟨λ ⟨D, h⟩, by exactI ⟨_, h, funext $ λ a, propext (to_bool_iff _).symm⟩,
- by rintro ⟨f, h, rfl⟩; exact ⟨by apply_instance, by simpa using h⟩⟩
+ by rintro ⟨f, h, rfl⟩; exact ⟨infer_instance, by simpa using h⟩⟩
 
 protected theorem not {p : α → Prop}
   (hp : computable_pred p) : computable_pred (λ a, ¬ p a) :=
```

Comment:
For some reason, line 169 (two lines below this) breaks when `by apply_instance` is changed to `infer_instance`. Weird.

Reply:
The reason is probably that the term generated by `by apply_instance` is the actual instance, while `infer_instance` has to be unfolded before it produces the instance. So they are not quite identical.